OpenSim/Sandbox/futureRigidContact.cpp

```diff
@@ -0,0 +1,87 @@
+#include <OpenSim/OpenSim.h>
+
+#define USE_RIGID_CONTACT
+
+using namespace OpenSim;
+using namespace SimTK;
+
+int main()
+{
+    // Physical parameters.
+    const double mass        = 1.;
+    const double radius      = 0.25;
+    const double stiffness   = 1.e7;
+    const double dissipation = 0.1;
+    const double mu_static   = 0.6;
+    const double mu_dynamic  = 0.4;
+    const double mu_viscous  = 0.01;
+    const double restitution = 0.5;
+
+    // Simulation parameters.
+    const double initialHeight   = 10.*radius;
+    const double initialHorizVel = 1.;
+    const double simDuration     = 0.15;
+    const double simStepSize     = 1.e-2;
+
+    // Create model with gravity.
+    Model *osimModel = new Model;
+    osimModel->setName("BallDrop");
+    osimModel->setGravity(Vec3(0, -9.81, 0));
+
+    // Create unconstrained ball.
+    OpenSim::Body ball("ball", mass, Vec3(0), mass*SimTK::Inertia::sphere(0.1));
+    ball.scale(Vec3(radius), false);
+    FreeJoint free("", osimModel->getGround().getName(), "ball");
+    osimModel->addBody(&ball);
+    osimModel->addJoint(&free);
```

Comment:
You will likely get a double-free error if these are not all dynamically allocated (though this comment is not relevant for pseudocode).